readinessProbe/readiness/server_test.go

```diff
@@ -12,67 +11,101 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-var _ = Describe("ManagedOCS Readiness Probe", func() {
-	// Define utility constants for object names and testing timeouts/durations and intervals.
-	const (
-		timeout  = time.Second * 10
-		duration = time.Second * 10
-		interval = time.Millisecond * 250
-	)
-
-	When("managedocs reports its components as \"not ready\"", func() {
-		BeforeEach(func() {
-			ctx := context.Background()
-
-			managedOCS := &v1.ManagedOCS{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      ManagedOCSName,
-					Namespace: TestNamespace,
+var _ = Describe("ManagedOCS readiness probe behavior", func() {
+	ctx := context.Background()
+
+	managedOCS := &v1.ManagedOCS{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ManagedOCSName,
+			Namespace: TestNamespace,
+		},
+	}
+
+	setupReadinessConditions := func(
+		storageClusterReady bool,
+		prometheusReady bool,
+		alertmanagerReady bool,
+	) {
+
+		var scStatus, prometheusStatus, alertmanagerStatus v1.ComponentState
+
+		if storageClusterReady == true {
+			scStatus = v1.ComponentReady
+		} else {
+			scStatus = v1.ComponentPending
+		}
+
+		if prometheusReady == true {
+			prometheusStatus = v1.ComponentReady
+		} else {
+			prometheusStatus = v1.ComponentPending
+		}
+
+		if alertmanagerReady == true {
+			alertmanagerStatus = v1.ComponentReady
+		} else {
+			alertmanagerStatus = v1.ComponentPending
+		}
+
+		Expect(k8sClient.Get(ctx, utils.GetResourceKey(managedOCS), managedOCS)).Should(Succeed())
```

Comment:
If you decided to assert inside the util (using Expect in this case) please use the WithOffset variants, if you will not use it the report errors will just point to the util and not the use-case.

[Suggestion] 2 other options:  
1. Do the get (and assert it) from the outside, send the `managedOCS` object to the util to be updated then do the update call (and assertion) from the outside

2. [My favourite] do the get and update in the util but do not assert just return the error in case of an error. On hte call site just assert the call to the util 
```
Expect(setupReadinessConditions).Should(Succeed())
```